API/Controllers/ProjectController.cs

```diff
@@ -1058,6 +1104,176 @@ public async Task<IActionResult> UnlinkInstitutionFromProjectAsync(int projectId
+
+            return Ok(mapper.Map<Project, ProjectResourceResult>(project));
+        }
+
+        /// <summary>
+        ///     Remove a category from a project
+        /// </summary>
+        /// <param name="projectId"></param>
+        /// <param name="categoryId"></param>
+        /// <response code="200">Category was removed from the project.</response>
+        /// <response code="401">User is not authorized to remove the category from the project.</response>
+        /// <response code="404">Project or category was not found.</response>
+        /// <response code="409">Project is not categorized with the category.</response>
+        /// <returns></returns>
+        [HttpDelete("category/{projectId}/{categoryId}")]
+        [Authorize]
+        [ProducesResponseType(typeof(ProjectResourceResult), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status409Conflict)]
+        public async Task<IActionResult> ProjectRemoveCategory(int projectId, int categoryId)
+        {
+            Project project = await projectService.FindAsync(projectId)
+                                                  .ConfigureAwait(false);
+            if(project == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to remove the category from the project.",
+                    Detail = "The project could not be found in the database.",
+                    Instance = "2CC94251-9103-4AAC-B461-F99939E78AD0"
+                };
+                return NotFound(problem);
+            }
+
+            Category category = await categoryService.FindAsync(categoryId)
+                                      .ConfigureAwait(false);
+
+            if(category == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to remove the category from the project.",
+                    Detail = "The category could not be found in the database.",
+                    Instance = "3E41B5DC-F78B-429B-89AB-1A98A6F65FDC"
+                };
+                return NotFound(problem);
+            }
+
+            User user = await HttpContext.GetContextUser(userService)
+                                         .ConfigureAwait(false);
+            bool isAllowed = await authorizationHelper.UserIsAllowed(user,
+                                                                     nameof(Defaults.Scopes.AdminProjectWrite),
+                                                                     nameof(Defaults.Scopes.InstitutionProjectWrite),
+                                                                     project.UserId);
+
+            if(project.UserId != user.Id && !isAllowed)
```

Comment:
Unauthorized should return if one or both conditions are false, right?

```suggestion
            if(!(project.UserId == user.Id || isAllowed))
```

Reply:
I think the correction you gave is very confusing. I read it as the following:

**If project userid is not user id AND not allowed, then fail.**
Yours, I would read like this:
**if not project user id is user id OR is allowed**
This makes it very hard to read to me.

In essence, it does the same. Look:

Mine:
userId = userId | isAllowed | result
---|---|---
false | false | true
false | true | false
true | false | false
true | true | false

Yours:
userId = userId | isAllowed | result
---|---|---
true | true | false
true | false | false
false | true | false
false | false | true

It does the same, i just think yours is harder to read.

Reply:
Correct me if I'm wrong but I think in your case both conditions should meet before it would enter the statement.
userId is not equal **AND** isAllowed = false. 

I think if only one of them meets, it should enter the statement.

@Maxvanhattum maybe second opinion?